commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,127 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+/**
+ * Created by misao on 16-Dec-15.
+ */
+public class ImageProcessing {
+
+    private Uri uri;
+    private Context context;
+
+    public ImageProcessing(Context context, Uri uri){
+        this.context = context;
+        this.uri = uri;
+    }
+
+    //Gets file path of image from its Uri
+    public String getFilePath(){
+        String filePath ="";
+        // Will return "image:x*"
+        String wholeID = DocumentsContract.getDocumentId(uri);
+
+        // Split at colon, use second item in the array
+        String id = wholeID.split(":")[1];
+        String[] column = { MediaStore.Images.Media.DATA };
+
+        // where id is equal to
+        String sel = MediaStore.Images.Media._ID + "=?";
+        Cursor cursor = context.getContentResolver().query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+                column, sel, new String[]{id}, null);
+
+        int columnIndex = cursor.getColumnIndex(column[0]);
+
+        if (cursor.moveToFirst()) {
+            filePath = cursor.getString(columnIndex);
+        }
+        cursor.close();
+
+        Log.d("Image", "File path: " + filePath);
+        return filePath;
+    }
+
+    //Extract GPS coords of image
+    public String getCoords(String filePath) {
```

Comment:
AFAIK the API request literally needs the coords to be submitted as a string though, so it seems like we would be converting it back to a string in the very next step? Apologies if I'm missing something.